src/main/kotlin/uk/gov/justice/digital/hmpps/sentenceplan/services/PlanService.kt

```diff
@@ -31,6 +32,17 @@ class PlanService(
     return planEntity.currentVersion!!
   }
 
+  fun lockPlan(planUuid: UUID): PlanVersionEntity {
+    val planEntity = planRepository.getPlan(planUuid)
+    planEntity.currentVersion?.status = CountersigningStatus.LOCKED_INCOMPLETE
+    planRepository.save(planEntity)
+
+    val newVersion = versionService.createNewPlanVersion(planEntity.currentVersion!!.uuid).apply { status = CountersigningStatus.UNSIGNED }
```

Comment:
I just changed this underneath you, if you call `versionService.alwaysCreateNewPlanVersion` that should work. Ditto any changes in mocks.

Reply:
Cool - will change

Reply:
rebased and fixed